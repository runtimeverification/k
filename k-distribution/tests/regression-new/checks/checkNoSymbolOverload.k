module CHECKNOSYMBOLOVERLOAD-SYNTAX
endmodule

module CHECKNOSYMBOLOVERLOAD
  syntax LVal
  syntax RVal
  syntax Val ::= LVal | RVal
  syntax Exp ::= Val

  syntax RVals ::= RVal [overload(arg)]
  syntax Vals  ::= Val  [overload(arg), symbol(val)]
  syntax Exps  ::= Exp  [overload(arg)]
endmodule
