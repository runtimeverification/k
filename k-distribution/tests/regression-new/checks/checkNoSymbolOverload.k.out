[Error] Compiler: Production would not be a KORE symbol and therefore cannot be overloaded. Add a `symbol(_)` attribute to the production.
	Source(checkNoSymbolOverload.k)
	Location(10,20,10,40)
	10 |	  syntax RVals ::= RVal [overload(arg)]
	   .	                   ^~~~~~~~~~~~~~~~~~~~
[Error] Compiler: Production would not be a KORE symbol and therefore cannot be overloaded. Add a `symbol(_)` attribute to the production.
	Source(checkNoSymbolOverload.k)
	Location(12,20,12,40)
	12 |	  syntax Exps  ::= Exp  [overload(arg)]
	   .	                   ^~~~~~~~~~~~~~~~~~~~
[Error] Compiler: Had 2 structural errors.
